pkg/freeze/docker/pause.go

```diff
@@ -0,0 +1,105 @@
+package docker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	dockerapi "github.com/docker/docker/client"
+)
+
+const defaultDockerUri = "unix:///var/run/docker.sock"
+const version = "1.19"
+
+var ErrNoNonQueueProxyPods = errors.New("no non queue-proxy containers found in pod")
+
+type CRI interface {
+	List(ctx context.Context, podUID string) ([]string, error)
+	Pause(ctx context.Context, container string) error
+	Resume(ctx context.Context, container string) error
+}
+
+type Docker struct {
+	cri CRI
+}
+
+// New return a FreezeThawer based on Docker.
+func New(c CRI) *Docker {
+	return &Docker{cri: c}
+}
+
+func NewCRI() (*DockerCRI, error) {
+	c, err := dockerapi.NewClientWithOpts(dockerapi.WithHost(defaultDockerUri),
+		dockerapi.WithVersion(version))
+	if err != nil {
+		return nil, err
+	}
+	return &DockerCRI{client: c}, nil
+}
+
+func (d *Docker) Freeze(ctx context.Context, podUID string) error {
```

Comment:
would it be possible (with some renaming :)) to use one top-level FreezeThawer for docker/crio/containerd and isolate all the differences to the CRI wrappers?